src/utils/accounts.js

```diff
@@ -111,9 +111,15 @@ export const isSmartWalletAccount = (account: ?Account): boolean => [
   ACCOUNT_TYPES.ETHERSPOT_SMART_WALLET,
 ].includes(account?.type);
 
-export const isKeyBasedAccount = (account: Account): boolean => account.type === ACCOUNT_TYPES.KEY_BASED;
+export const isAccountType = (account: ?Account, type: string): boolean => account?.type === type;
 
-export const isArchanovaAccount = (account: ?Account): boolean => account?.type === ACCOUNT_TYPES.SMART_WALLET;
+export const isArchanovaAccount = (
+  account: ?Account,
+): boolean => isAccountType(account, ACCOUNT_TYPES.SMART_WALLET);
```

Comment:
Consider renaming the const for clarity
```suggestion
): boolean => isAccountType(account, ACCOUNT_TYPES.ARCHANOVA_SMART_WALLET);
```

Reply:
It's actually renamed in other PR, leaving unchanged here 👍